web/src/saved_snippets.ts

```diff
@@ -0,0 +1,51 @@
+import type {Option} from "./dropdown_widget";
+import {$t} from "./i18n";
+import * as util from "./util";
+
+export type SavedSnippet = {
+    id: number;
+    title: string;
+    content: string;
+    date_created: number;
+};
+
+export const ADD_SAVED_SNIPPET_OPTION_ID = -1;
+let my_saved_snippets: SavedSnippet[] = [];
```

Comment:
Use a map data structure keyed by the IDs; the array format is only for network transit. Half of your queries are things that would be an O(1) lookup in a map.

(At least one `get_saved_snippets` caller is iterating through the list to find a single item by ID, and `snippet` is also very strange)

Reply:
OK, used map instead of an array for storing saved snippets on the client side and changed `get_saved_snippets` function to `get_saved_snippet_by_id`.